src/components/ItemEditorPage/RightPanel/RightPanel.tsx

```diff
@@ -110,14 +110,32 @@ export default class RightPanel extends React.PureComponent<Props> {
                 label={t('item_editor.right_panel.replaces')}
                 value={item.data.replaces}
                 options={Object.values(WearableCategory).map(value => ({ value, text: t(`wearable.category.${value}`) }))}
-                onChange={replaces => this.handleChange({ ...item, data: { ...item.data, replaces } })}
+                onChange={replaces =>
+                  this.handleChange({
+                    ...item,
+                    data: {
+                      ...item.data,
+                      replaces,
+                      representations: item.data.representations.map(representation => ({ ...representation, overrideReplaces: replaces }))
+                    }
+                  })
+                }
               />
               <MultiSelect<WearableCategory>
                 itemId={item.id}
                 label={t('item_editor.right_panel.hides')}
                 value={item.data.hides}
                 options={Object.values(WearableCategory).map(value => ({ value, text: t(`wearable.category.${value}`) }))}
-                onChange={hides => this.handleChange({ ...item, data: { ...item.data, hides } })}
+                onChange={hides =>
+                  this.handleChange({
+                    ...item,
+                    data: {
+                      ...item.data,
+                      hides,
+                      representations: item.data.representations.map(representation => ({ ...representation, overrideHides: hides }))
+                    }
+                  })
+                }
```

Comment:
I didn't add this one as a class method because i need `item` which is not available at the class level. Also couldn't reuse much of the logic because the two handlers modify `item.data` in different ways (one sets `item.data.replaces` and the other `item.data.hides`) and same thing for each `representation` (one sets `representation.overrideHides` and the other `representation.overrideReplaces`)